app/controllers/sequences_controller.rb

```diff
@@ -44,6 +44,38 @@ def show
       @lineages = sequence.lineages(equate_il, true).to_a
     end
 
+    # create list for all ec_cross_reference numbers
+    ecnumbers = []
```

Comment:
Use a set instead for linear performance. Calling uniq on array will probably have a quadratic cost.
Another option is to first use an array (and used a double map + flatten instead of the double each below) and then add everything to a map.


_[Original comment](https://github.ugent.be/unipept/unipept/pull/528#discussion_r3381) by @bmesuere on Wed Oct 21 2015 at 16:42._